src/pages/UserMyVisits.jsx

```diff
@@ -1,11 +1,158 @@
-import React from 'react';
-import { Layout } from '../components';
-import { Typography } from '@mui/material';
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+
+import Grid from '@mui/material/Grid';
+import EventAvailableIcon from '@mui/icons-material/EventAvailable';
+import isSameDay from 'date-fns/isSameDay';
+
+import { Layout, DatePicker, CustomButton } from '../components';
+import { paths } from '../config/paths';
+import { db, auth } from '../config/firebase';
+import { doc, deleteDoc } from 'firebase/firestore';
 
 export const UserMyVisits = () => {
+  const [loading, setLoading] = useState(true);
+  const [visits, setVisits] = useState([]);
+  const userId = auth.currentUser.auth.lastNotifiedUid;
+  const [visitsDates, setVisitsDates] = useState([]);
+  const [arrayOfVisits, setArrayOfVisits] = useState([]);
+  const [date, setDate] = useState(new Date());
+  const [selected, setSelected] = useState([]);
+
+  useEffect(() => {
+    const getVisitsFromFirebase = [];
+    const visitsArray = [];
+    db.collection('visits').onSnapshot((querySnapshot) => {
+      querySnapshot.forEach((doc) => {
+        getVisitsFromFirebase.push({
+          ...doc.data(),
+
+          key: doc.id,
+        });
+      });
+      getVisitsFromFirebase.forEach((visit) => {
+        if (visit.uid === userId) {
+          visitsArray.push(visit);
+        }
+      });
+      setVisits(visitsArray);
+      setLoading(false);
+    });
+  }, [loading, userId]);
+
+  useEffect(() => {
+    const dates = [];
+    visits.forEach((visit) => {
+      dates.push(new Date(visit.date.seconds * 1000 + visit.date.nanoseconds / 1000000));
+      return dates;
+    });
+    setVisitsDates(dates);
+  }, [visits]);
+
+  useEffect(() => {
+    const info = visitsDates.find((e) => isSameDay(date, e));
+    setSelected([info]);
+  }, [date, visitsDates]);
+
+  useEffect(() => {
+    const visitsRenderArray = [];
+    visits.forEach((el) => {
+      selected.forEach((selection) => {
+        if (selection !== undefined) {
+          if (
+            selection.toLocaleDateString({ year: 'numeric', month: 'long', day: 'numeric' }) ===
+            new Date(el.date.seconds * 1000 + el.date.nanoseconds / 1000000).toLocaleDateString({
+              year: 'numeric',
+              month: 'long',
+              day: 'numeric',
+            })
+          ) {
+            if (visitsRenderArray.includes(el) === false) visitsRenderArray.push(el);
+            return visitsRenderArray;
+          }
+        }
+      });
+    });
+    setArrayOfVisits(visitsRenderArray);
+  }, [visits, selected]);
+
+  if (loading) {
+    return <h1>loading data...</h1>;
+  }
+
+  const deleteVisit = (key) => {
+    deleteDoc(doc(db, 'visits', `${key}`));
+    setVisits([]);
```

Comment:
Maybe you also need to remove an item from arrayOfVisits or visitsDates?

Reply:
Looks like, moving arrays inside functions fixed the issue. I had some arrays as global variables and this caused the weird behavior. Thanks for pointing this out, it helped me fixed the problem! :)